### ISSUE TITLE
Announcement Center dashboard doesn't work if user status is disabled

### ISSUE THREAD

**Issue by AJDurant** [nextcloud/announcementcenter#279]
This seems to have already been fixed in 75e6372 but that has not been released for NC20

### Steps to reproduce
1. Add announcementcenter 4.0.0 to Nextcloud 20
2. Disable user_status app
3. Add announcements to dashboard
4. announcements fails to load

### Expected behaviour
Announcements should load on dashboard

### Actual behaviour
Announcements fail to load, spinner shows permanently

## Server configuration detail

**Operating system:** Linux 5.4.0-64-generic #72-Ubuntu SMP Fri Jan 15 10:27:54 UTC 2021 x86_64

**Webserver:** Apache/2.4.41 (Ubuntu) (apache2handler)

**Database:** mysql 10.3.25

**PHP version:** 

7.4.3
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, sodium, apache2handler, mysqlnd, PDO, xml, apcu, bcmath, bz2, calendar, ctype, curl, dom, mbstring, FFI, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, imagick, imap, intl, json, ldap, exif, mysqli, pdo_mysql, pdo_sqlite, apc, posix, readline, redis, shmop, SimpleXML, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tokenizer, xmlreader, xmlwriter, xsl, zip, Phar, Zend OPcache

**Nextcloud version:** 20.0.6 - 20.0.6.1

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** unknown

<details><summary>Signing status</summary>

Array
(
)

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.6.0
 - activity: 2.13.4
 - admin_audit: 1.10.0
 - announcementcenter: 4.0.0
 - apporder: 0.11.0
 - bruteforcesettings: 2.0.1
 - bulkgroups: 0.0.1
 - cloud_federation_api: 1.3.0
 - companyhome: 0.0.7
 - customermanager: 0.1.3
 - dashboard: 7.0.0
 - dav: 1.16.2
 - external: 3.7.2
 - federatedfilesharing: 1.10.2
 - files: 1.15.0
 - files_rightclick: 0.17.0
 - files_sharing: 1.12.2
 - files_trackdownloads_custom: 1.7.1
 - files_trashbin: 1.10.1
 - files_versions: 1.13.0
 - firstrunwizard: 2.9.0
 - groupfolders: 8.2.0
 - impersonate: 1.7.0
 - issuetemplate: 0.7.0
 - logreader: 2.5.0
 - lookup_server_connector: 1.8.0
 - notifications: 2.8.0
 - oauth2: 1.8.0
 - password_policy: 1.10.1
 - provisioning_api: 1.10.0
 - serverinfo: 1.10.0
 - settings: 1.2.0
 - sharebymail: 1.10.0
 - support: 1.3.0
 - theming: 1.11.0
 - twofactor_backupcodes: 1.9.0
 - updatenotification: 1.10.0
 - user_ldap: 1.10.2
 - viewer: 1.4.0
 - workflowengine: 2.2.0
Disabled:
 - comments
 - contactsinteraction
 - encryption
 - federation
 - files_external
 - files_pdfviewer
 - files_videoplayer
 - nextcloud_announcements
 - notes
 - photos
 - privacy
 - recommendations
 - socialsharing_email
 - survey_client
 - systemtags
 - text
 - user_status
 - weather_status

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "***REMOVED SENSITIVE VALUE***",
        "***REMOVED SENSITIVE VALUE***"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "20.0.6.1",
    "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "htaccess.RewriteBase": "\/",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379,
        "timeout": 3
    },
    "default_locale": "en_GB",
    "default_language": "en_GB",
    "defaultapp": "apporder",
    "knowledgebaseenabled": false,
    "activity_use_cached_mountpoints": true,
    "enable_previews": false,
    "mail_smtpmode": "sendmail",
    "mail_smtpsecure": "tls",
    "mail_sendmailmode": "smtp",
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpport": "25",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "mail_smtpauthtype": "LOGIN",
    "ldapIgnoreNamingRules": false,
    "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
    "loglevel": 0,
    "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
    "remember_login_cookie_lifetime": 0,
    "skeletondirectory": "",
    "logfile": "\/var\/log\/nexcloud\/nextcloud.log",
    "log.condition": {
        "apps": [
            "admin_audit"
        ]
    },
    "customclient_desktop": "",
    "customclient_android": "",
    "customclient_ios": "",
    "customclient_ios_appid": "",
    "apps_paths": [
        {
            "path": "\/var\/www\/nextcloud\/apps",
            "url": "\/apps",
            "writable": true
        },
        {
            "path": "\/var\/www\/nextcloud\/apps_custom",
            "url": "\/apps_custom",
            "writable": false
        }
    ],
    "maintenance": false,
    "theme": "",
    "updater.secret": "***REMOVED SENSITIVE VALUE***",
    "app_install_overwrite": [
        "companyhome",
        "customermanager",
        "files_trackdownloads_custom",
        "bulkgroups"
    ],
    "debug": true
}
```
</details>

**Are you using external storage, if yes which one:** 

**Are you using encryption:** 

**Are you using an external user-backend, if yes which one:** ActiveDirectory

<details><summary>LDAP configuration (delete this par if not used)</summary>

```

```
</details>

## Client configuration

**Browser:** Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36

**Operating system:** 

## Logs

<details><summary>Web server error log</summary>

```
Insert your web server log here 
```
</details>

<details><summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

<details><summary>Browser log</summary>

index.js:165 Uncaught TypeError: Cannot read property 'substring' of undefined
    at r (index.js:165)
    at Module.<anonymous> (dashboard.js:33)
    at n (bootstrap:19)
    at bootstrap:83
    at announcementcenter-dashboard.js:1

which remaps to:
announcementcenter-dashboard.js:formatted:6735 Uncaught TypeError: Cannot read property 'substring' of undefined
    at r (announcementcenter-dashboard.js:formatted:6735)
    at Module.<anonymous> (announcementcenter-dashboard.js:formatted:43318)
    at n (announcementcenter-dashboard.js:formatted:11)
    at announcementcenter-dashboard.js:formatted:71
    at announcementcenter-dashboard.js:formatted:72
</details>

**Comment by nickvergessen** [comment]
Ah, thanks for the detail about user_status being disbaled. This is what I missed so far.

**Comment by nickvergessen** [comment]
Fixed with #280 
v4.0.1 is published to the appstore
